src/extension/conversation/vscode-node/languageModelAccess.ts

```diff
@@ -153,19 +153,29 @@ export class LanguageModelAccess extends Disposable implements IExtensionContrib
 			// Counting tokens requires instantiating the tokenizers, which makes this process use a lot of memory.
 			// Let's cache the results across extension activations
 			const baseCount = await this._promptBaseCountCache.getBaseCount(endpoint);
-			let multiplierString = endpoint.multiplier !== undefined ? `${endpoint.multiplier}x` : undefined;
+			let modelDetail = endpoint.multiplier !== undefined ? `${endpoint.multiplier}x` : undefined;
+
 			if (endpoint.model === AutoChatEndpoint.id) {
-				multiplierString = 'Variable';
+				modelDetail = 'Variable';
+			}
+			let modelName = endpoint.name;
+			if (endpoint.customModel) {
+				const customModel = endpoint.customModel;
+				modelDetail = customModel.ownerName;
+				modelDescription = customModel.keyName;
+				modelCategory = { label: localize('languageModelHeader.custom_models', "Custom Models"), order: 1 };
+
+				modelName = endpoint.name.replace(` · ${customModel.keyName}`, '');
```

Comment:
Can we convince CAPI to not append key names like this? If they're worried about older clients than an API version would make sense here. I really dislike arbitrary string manipulation

Reply:
yeah I will do the change in CAPI as soon as this update goes out instead! good call, keep this clean!